### PR TITLE
Add table_aws_efs_file_system closes #134

### LINKED ISSUE #134: Add table_aws_efs_file_system
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.37.24/service/efs

### PR DESCRIPTION
**Pull request by Paulami30**

### CODE REVIEW

File: aws/service.go [hunk 1]
```diff
@@ -202,6 +203,29 @@ func Ec2Service(ctx context.Context, d *plugin.QueryData, region string) (*ec2.E
 	return svc, nil
 }
 
+// EFSService returns the service connection for AWS EFS File System service
```
Comment (Subhajit97) [inline aws/service.go:206]:
```suggestion
// EFSService returns the service connection for AWS Elastic File System service
```

File: aws/table_aws_efs_file_system.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/efs"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+)
+
+// TABLE DEFINITION
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:15]:
```suggestion
//// TABLE DEFINITION
```

File: aws/table_aws_efs_file_system.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("FileSystemArn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+// LIST FUNCTION
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:140]:
```suggestion
//// LIST FUNCTION
```

File: aws/table_aws_efs_file_system.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+				d.StreamListItem(ctx, fileSystem)
+			}
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+// HYDRATE FUNCTIONS
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:170]:
```suggestion
//// HYDRATE FUNCTIONS
```

File: aws/table_aws_efs_file_system.go [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+		return nil, err
+	}
+
+	// Build param
+	param := &efs.DescribeFileSystemPolicyInput{
+		FileSystemId: fileSystem.FileSystemId,
+	}
+
+	fileSystemPolicy, err := svc.DescribeFileSystemPolicy(param)
+
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:230]:
```suggestion
```

File: aws/table_aws_efs_file_system.go [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+			if a.Code() == "PolicyNotFound" {
+				return nil, nil
+			}
+			return nil, err
+		}
+	}
+	return fileSystemPolicy, nil
+}
+
+// TRANSFORM FUNCTIONS
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:242]:
```suggestion
//// TRANSFORM FUNCTIONS
```

File: aws/table_aws_efs_file_system.go [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+func elasticFileSystemTurbotData(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	fileSystemTag := d.HydrateItem.(*efs.FileSystemDescription)
+
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:246]:
```suggestion

if fileSystemTag.Tags == nil {
  return nil, nil
}

```

File: aws/table_aws_efs_file_system.go [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+			KeyColumns:        plugin.SingleColumn("file_system_id"),
+			ShouldIgnoreError: isNotFoundError([]string{"FileSystemNotFound"}),
+			Hydrate:           getElasticFileSystem,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listElasticFileSystem,
+		},
+		GetMatrixItem: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:31]: Add `name` column
```
// You can add tags to a file system, including a Name tag. For more information,
// see CreateFileSystem. If the file system has a Name tag, Amazon EFS returns
// the value in this field.
Name *string `type:"string"`
```

File: aws/table_aws_efs_file_system.go [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("FileSystemId"),
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:122]: Set the `Name` as title, if not null.

File: docs/tables/aws_efs_file_system.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+      jsonb_array_elements_text(s -> 'Action') as a,
+      jsonb_array_elements_text(
+        s -> 'Condition' -> 'Bool' -> 'aws:securetransport'
+      ) as ssl
+    where
+      p = '*'
+      and s ->> 'Effect' = 'Deny'
+      and ssl :: bool = false
+  );
+```
```
Comment (Subhajit97) [inline docs/tables/aws_efs_file_system.md:79]:
```suggestion
```

```

Comment (Subhajit97) [inline docs/tables/aws_efs_file_system.md:79]: Add a blank line at the end

File: aws-test/tests/aws_efs_file_system/test-hydrate-query.sql [hunk 11]
```diff
@@ -0,0 +1,3 @@
+select file_system_id, akas, tags, title
```
Comment (Subhajit97) [inline aws-test/tests/aws_efs_file_system/test-hydrate-query.sql:1]:
```suggestion
select file_system_id, policy, akas, tags, title
```

File: aws/table_aws_efs_file_system.go (hunk 9 above)
Comment (Paulami30) [reply aws/table_aws_efs_file_system.go:122]: I think FileSystemId is better as title than name. The name can also be duplicate

File: aws-test/tests/aws_efs_file_system/test-list-query.sql [hunk 12]
```diff
@@ -0,0 +1,3 @@
+select akas, file_system_id, encrypted, performance_mode, title
```
Comment (Subhajit97) [inline aws-test/tests/aws_efs_file_system/test-list-query.sql:1]: Add test for `tags`
```suggestion
select akas, file_system_id, encrypted, performance_mode, title, tags
```

File: aws/table_aws_efs_file_system.go [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+
+// TABLE DEFINITION
+
+func tableAwsElasticFileSystem(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_efs_file_system",
+		Description: "AWS Elastic File System",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("file_system_id"),
+			ShouldIgnoreError: isNotFoundError([]string{"FileSystemNotFound"}),
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:23]: Handle validation error. It should return error for these cases
image

File: aws/table_aws_efs_file_system.go [hunk 14]
```diff
@@ -0,0 +1,256 @@
...
+			{
+				Name:        "number_of_mount_targets",
+				Description: "The current number of mount targets that the file system has.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "size_in_bytes",
+				Description: "The latest known metered size (in bytes) of data stored in the file system.",
+				Type:        proto.ColumnType_JSON,
+			},
```
Comment (Subhajit97) [inline aws/table_aws_efs_file_system.go:70]: Move all the JSON type columns at the end

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
Make suggested changes, and add the tests logs.

**Comment by Paulami30** [comment]
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_efs_file_system []

PRETEST: tests/aws_efs_file_system

TEST: tests/aws_efs_file_system
Running terraform
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.alternate: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_efs_file_system.named_test_resource: Creating...
aws_efs_file_system.named_test_resource: Still creating... [10s elapsed]
aws_efs_file_system.named_test_resource: Still creating... [20s elapsed]
aws_efs_file_system.named_test_resource: Still creating... [30s elapsed]
aws_efs_file_system.named_test_resource: Creation complete after 40s [id=fs-d88ba42e]
data.template_file.resource_aka: Refreshing state...
aws_efs_file_system_policy.efs_file_system_policy: Creating...
aws_efs_file_system_policy.efs_file_system_policy: Creation complete after 4s [id=fs-d88ba42e]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = arn:aws:elasticfilesystem:us-east-1:352685002396:file-system/fs-d88ba42e
resource_id = fs-d88ba42e
resource_name = turbottest73964

Running SQL query: test-get-query.sql
[
  {
    "encrypted": true,
    "file_system_id": "fs-d88ba42e",
    "performance_mode": "maxIO",
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest73964"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "arn:aws:elasticfilesystem:us-east-1:352685002396:file-system/fs-d88ba42e"
    ],
    "file_system_id": "fs-d88ba42e",
    "policy": {
      "Id": "test_policy",
      "Statement": [
        {
          "Action": [
            "elasticfilesystem:ClientMount",
            "elasticfilesystem:ClientWrite"
          ],
          "Condition": {
            "Bool": {
              "aws:SecureTransport": "true"
            }
          },
          "Effect": "Allow",
          "Principal": {
            "AWS": "*"
          },
          "Resource": "arn:aws:elasticfilesystem:us-east-1:352685002396:file-system/fs-d88ba42e",
          "Sid": "__default_policy_ID"
        }
      ],
      "Version": "2012-10-17"
    },
    "tags": {
      "name": "turbottest73964"
    },
    "title": "fs-d88ba42e"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "arn:aws:elasticfilesystem:us-east-1:352685002396:file-system/fs-d88ba42e"
    ],
    "encrypted": true,
    "file_system_id": "fs-d88ba42e",
    "performance_mode": "maxIO",
    "tags": {
      "name": "turbottest73964"
    },
    "title": "fs-d88ba42e"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

POSTTEST: tests/aws_efs_file_system

TEARDOWN: tests/aws_efs_file_system

SUMMARY:

1/1 passed.

**Review by Subhajit97 — DISMISSED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**